tests/e2e/idps_test.go

```diff
@@ -0,0 +1,82 @@
+package e2e
+
+import (
+	"fmt"
+	"net/http"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	conn "github.com/terraform-redhat/terraform-provider-rhcs/tests/ci"
+	"github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/cms"
+	con "github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/constants"
+	exe "github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/exec"
+)
+
+var _ = Describe("TF Test", func() {
+	Describe("Identity Providers test cases", func() {
+
+		// all identity providers - declared for future cases
+		type IDPServices struct {
+			htpasswd,
+			github,
+			gitlab,
+			google,
+			ldap,
+			openid exe.IDPService
+		}
+
+		var idpService IDPServices
+
+		Describe("Htpasswd IDP test cases", func() {
+			var htpasswdMap = []interface{}{map[string]string{}}
+			var htpasswdUsername, htpasswdPassword string
+
+			BeforeEach(func() {
+
+				htpasswdUsername = "jacko"
+				htpasswdPassword = "1q2wFe4rpoe2318"
+				htpasswdMap = []interface{}{map[string]string{"username": htpasswdUsername, "password": htpasswdPassword}}
+				idpService.htpasswd = *exe.NewIDPService(con.HtpasswdDir) // init new htpasswd service
+			})
+
+			AfterEach(func() {
+				err := idpService.htpasswd.Destroy()
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			Context("Author:smiron-High-OCP-63151 @OCP-63151 @smiron", func() {
+				It("OCP-63151 - Provision HTPASSWD IDP against cluster using TF", func() {
+					By("Create htpasswd idp for an existing cluster")
+
+					idpParam := &exe.IDPArgs{
+						Token:         token,
+						ClusterID:     clusterID,
+						Name:          "htpasswd-idp-test",
+						HtpasswdUsers: htpasswdMap,
+					}
+					err := idpService.htpasswd.Create(idpParam, "-auto-approve", "-no-color")
+					Expect(err).ToNot(HaveOccurred())
+					idpID, _ := idpService.htpasswd.Output()
+
+					By("List existing HtpasswdUsers and compare to the created one")
+					htpasswdUsersList, _ := cms.ListHtpasswdUsers(conn.RHCSConnection, clusterID, idpID.ID)
+					Expect(htpasswdUsersList.Status()).To(Equal(http.StatusOK))
+					respUserName, _ := htpasswdUsersList.Items().Slice()[0].GetUsername()
+					Expect(respUserName).To(Equal(htpasswdUsername))
+
+					By("Login with created htpasswd idp")
+					getresp, err := cms.RetrieveClusterDetail(conn.RHCSConnection, clusterID)
+					Expect(err).ToNot(HaveOccurred())
+					server := getresp.Body().API().URL()
+
+					userLoginCMD := exe.GenerateOCLoginCMD(server, htpasswdUsername, htpasswdPassword, clusterID, "--insecure-skip-tls-verify")
+					output, err := exe.RetryCMDRun(userLoginCMD, 5)
```

Comment:
A suggestion:
Why not having one function "OcLogin" that wraps those two functions
It can be function in struct and the server provided in the constructor

Than in the future will add more actions to this struct, using the CMD commands

Reply:
hmm let me see if I understood you correctly - 
the suggestion is to create a structure for future CMD operations, and one of the attributes
of this structure will be the function OcLogin that will combine those 2 functions?

Reply:
Yes, something like this.
The idea is to start creating the infrastructure for oc actions
If you prefer to do this in different PR I'm also ok with this

Reply:
ack, so Iv'e changed part of it based on your comment, 
and the struct itself will be implemented in a different PR :D